### PR TITLE
Small cleanup related to resolution proof graph

### PR DESCRIPTION
**Pull request by blishko**
This PR removes some of the unused fields and methods in `ProofGraph` and also removes the last usage of implication verification using external tool.
Internal verification should be sufficient in my opinion.

### CODE REVIEW

File: src/proof/PGCheck.cc [hunk 1]
```diff
@@ -21,102 +21,55 @@ along with Periplo. If not, see <http://www.gnu.org/licenses/>.
...
+
+    vec<PTRef> clauses;
+	for (clauseid_t leaf_id : leaves_ids) {
+        auto const & clause = getNode(leaf_id)->getClause();
+        vec<PTRef> lits;
+        for (Lit l : clause) {
+            lits.push(termMapper.litToPTRef(l));
+        }
+        clauses.push(logic_.mkOr(std::move(lits)));
 	}
```
Comment (aehyvari) [inline src/proof/PGCheck.cc:68]: This looks like an indentation situation.

File: src/proof/PGCheck.cc [hunk 2]
```diff
@@ -21,102 +21,55 @@ along with Periplo. If not, see <http://www.gnu.org/licenses/>.
...
+            } else {
+                proofleaves.push_back(id);
+            }
+        }
+    } while (!q.empty());
+
+    vec<PTRef> clauses;
+	for (clauseid_t leaf_id : leaves_ids) {
+        auto const & clause = getNode(leaf_id)->getClause();
+        vec<PTRef> lits;
```
Comment (aehyvari) [inline src/proof/PGCheck.cc:63]: Maybe add `lits.capacity(clauses.size)`?

File: src/proof/PGCheck.cc [hunk 3]
```diff
@@ -21,102 +21,55 @@ along with Periplo. If not, see <http://www.gnu.org/licenses/>.
...
 
 	std::vector<clauseid_t> proofleaves;
-	clauseid_t id,id1,id2;
 	std::vector<clauseid_t> q;
-	ProofNode * node = NULL;
-	std::vector<unsigned>* visited_count_ = new std::vector<unsigned>(getGraphSize(),0);
-	std::vector<unsigned>& visited_count = *visited_count_;
-	// Building DFS vector
+	std::vector<unsigned> visited_count(getGraphSize(), 0u);
 	q.push_back(getRoot()->getId());
```
Comment (aehyvari) [inline src/proof/PGCheck.cc:36]: There seems to be an indentation problem.

**Review by aehyvari — COMMENTED** [review]
There are a few places where the changed code uses correctly spaces and old code uses tabs.  It looks like my clion fixes these problems visually, but they show up on GitHub and at least on my tab settings on vim.

File: src/proof/PGCheck.cc (hunk 2 above)
Comment (blishko) [reply src/proof/PGCheck.cc:62]: Added.

File: src/proof/PGCheck.cc (hunk 1 above)
Comment (blishko) [reply src/proof/PGCheck.cc]: Fixed.

File: src/proof/PGCheck.cc (hunk 3 above)
Comment (blishko) [reply src/proof/PGCheck.cc]: Fixed.

**Review by aehyvari — APPROVED** [review]
LGTM!

**Comment by blishko** [comment]
It looks like the whole file could use reformatting.
Maybe I'll just do that.

**Comment by aehyvari** [comment]
> It looks like the whole file could use reformatting.
> Maybe I'll just do that.

Sure, go ahead if you feel like that!

**Pull request merged**